apps/gdalalg_raster_calc.cpp

```diff
@@ -602,6 +615,35 @@ bool GDALRasterCalcAlgorithm::RunImpl(GDALProgressFunc pfnProgress,
         return false;
     }
 
+    if (!m_skipCheckExpression)
+    {
+        const bool bIsVRT =
+            m_format == "VRT" ||
+            (m_format.empty() &&
+             EQUAL(
+                 CPLGetExtensionSafe(m_outputDataset.GetName().c_str()).c_str(),
```

Comment:
Why two different methods for checking extensions?

Reply:
Yeah, that's a bit annoying. We should probably have a method that takes both the format and filename arguments and return the format. I'll let that for another PR